core/src/test/java/org/pentaho/platform/dataaccess/datasource/api/MetadataServiceTest.java

```diff
@@ -678,5 +680,32 @@ public void createNewByteArrayInputStream() throws Exception {
     assertNotNull( stream );
   }
 
+  @Test
+  public void testExtractXmiFromZip() throws Exception {
+    String path = getClass().getResource( "import_metadata.zip" ).getPath();
```

Comment:
do you need both files of "import_metadata.zip" or just one:
resources/import_metadata.zip
resources/org/pentaho/platform/dataaccess/datasource/api/import_metadata.zip